sdb/commands/zfs/zio.py

```diff
@@ -124,6 +124,14 @@ def from_zio(self, zio: drgn.Object) -> Iterable[drgn.Object]:
                 yield from self.from_zio(c.zl_parent)
         self.level -= 1
 
+    @staticmethod
+    def zio_has_parents(zio: drgn.Object) -> bool:
+        parent_list = zio.io_parent_list.list_head.address_of_()
+        first_parent = parent_list.next
```

Comment:
Can you just check to see if `parent_list` is not NULL?

Reply:
`io_parent_list` is not a pointer, it's a `list_t`, so it can't be NULL.

Reply:
I was wondering about `parent_list` which comes from `address_of_()`. I thought that returned a pointer to an Object.

Reply:
Yes, but `&zio->io_parent_list.list_head` can't be NULL.  If the zio was NULL then it would be `NULL + offsetof(zio_t, io_parent_list) + offsetof(list_t, list_head)` which is a pointer but not NULL (it's around 0x240).